spyder/plugins/plots/widgets/main_widget.py

```diff
@@ -0,0 +1,468 @@
+        if shellwidget_id in self._shellwidgets:
+            fig_browser = self._shellwidgets.pop(shellwidget_id)
+            self.remove_widget(fig_browser)
+            fig_browser.close()
+
+    def set_shellwidget(self, shellwidget):
+        shellwidget_id = id(shellwidget)
+        if shellwidget_id in self._shellwidgets:
+            fig_browser = self._shellwidgets[shellwidget_id]
+            fig_browser.setup(self._options)
+            self.set_current_widget(fig_browser)
+
+    # --- API: Main widget
+    # ------------------------------------------------------------------------
+    def show_figure_menu(self, qpoint):
+        self._right_clicked_thumbnail = None
+        widget = self.current_widget()
+        if widget:
+            self.get_menu(PluginMainWidgetMenus.Context).popup(qpoint)
+
+    def show_thumbnail_menu(self, qpoint, thumbnail):
+        self._right_clicked_thumbnail = thumbnail
+        widget = self.current_widget()
+        if widget:
+            self.get_menu(PluginMainWidgetMenus.Context).popup(qpoint)
+
+    def save_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            if self._right_clicked_thumbnail is None:
+                widget.thumbnails_sb.save_current_figure_as()
+            else:
+                widget.thumbnails_sb.save_thumbnail_figure_as(
+                    self._right_clicked_thumbnail)
+
+    def save_all_plots(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.save_all_figures_as()
+
+    def copy_image(self):
+        widget = self._stack.currentWidget()
+        if widget and widget.figviewer and widget.figviewer.figcanvas.fig:
+            if self._right_clicked_thumbnail is None:
+                widget.figviewer.figcanvas.copy_figure()
+            else:
+                self._right_clicked_thumbnail.canvas.copy_figure()
+
+    def remove_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            if self._right_clicked_thumbnail is None:
+                widget.thumbnails_sb.remove_current_thumbnail()
+            else:
+                widget.thumbnails_sb.remove_thumbnail(
+                    self._right_clicked_thumbnail)
+
+        self.update_actions()
+
+    def remove_all_plots(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.remove_all_thumbnails()
+
+        self.update_actions()
+
+    def previous_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.go_previous_thumbnail()
+
+    def next_plot(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.thumbnails_sb.go_next_thumbnail()
+
+    def zoom_in(self):
+        widget = self._stack.currentWidget()
+        if widget:
+            widget.zoom_in()
+
+    def zoom_out(self):
+        widget = self._stack.currentWidget()
```

Comment:
Missing docstring and `self.current_widget()`

Reply:
Fixed